napari_allencell_annotator/view/images_view.py

```diff
@@ -111,11 +111,19 @@ def _connect_slots(self) -> None:
         self.delete.clicked.connect(self._handle_delete_clicked)
         self.file_widget.files_selected.connect(self._toggle_delete_button_text)
 
-        self._annotator_model.image_changed.connect(self._display_img)
+        self._annotator_model.annotation_recorded.connect(self._display_img)
+        self._annotator_model.annotation_started_changed.connect(self._handle_annotation_started)
         self._annotator_model.image_count_changed.connect(self._handle_image_count_changed)
         self._annotator_model.images_shuffled.connect(self._handle_shuffle_ui)
         self._annotator_model.images_shuffled.connect(self.viewer.clear_layers)
 
+    def _handle_annotation_started(self) -> None:
+        """
+        Display the current image when annotation starts.
+        """
+        if self._annotator_model.is_annotation_started():
+            self._display_img()
+
```

Comment:
fix comment

Reply:
I was confused. This comment is correct. This method is called when annotation starts. When annotations are recorded, the `annotation_recorded` signal is subscribed directly by `_display_img()`. Therefore, this method has nothing to do with annotations being recorded.

Reply:
can you explain what this signal and method does ( and why it was added to this pr)? im a bit confused then

Reply:
previously _display_img is subscribed to the image_changed signal, so it displays a new image every time we set a new current image. However, now it's subscribed to the annotation_recorded signal, which is emitted only after annotations have been recorded. As a result, when we set a new current image without saving annotations, such as when we start annotation, the plugin would not display the image. This method handles this by subscribing to the annotation_started signal and display the image if annotation has been started.